elyra/pipeline/validation.py

```diff
@@ -165,17 +165,32 @@ def _validate_pipeline_structure(self, pipeline: dict, response: ValidationRespo
         elif 'version' not in primary_pipeline['app_data']:
             response.add_message(severity=ValidationSeverity.Error, message_type="invalidPipeline",
                                  message="Pipeline app_data 'version' is missing from primary pipeline")
-        elif not int(primary_pipeline['app_data']['version']) > 0:
+
+        # validate pipeline version compatibility
+        pipeline_version = int(primary_pipeline['app_data']['version'])
+        if pipeline_version not in range(PIPELINE_CURRENT_VERSION + 1):
             response.add_message(severity=ValidationSeverity.Error, message_type="invalidPipeline",
                                  message="Primary pipeline version field has an invalid value")
 
+        elif pipeline_version < PIPELINE_CURRENT_VERSION:
+            # Pipeline needs to be migrated
+            response.add_message(severity=ValidationSeverity.Error, message_type="invalidPipeline",
+                                 message=f'Pipeline version {pipeline_version} is out of date and needs to be migrated '
+                                         f'using the Elyra pipeline editor.')
```

Comment:
Might want to add something like the following to provide the caller with as much contextual information as possible

```
...,
data = {
  supported_version: PIPELINE_CURRENT_VERSION,
  detected_version: pipeline_version
}
```